keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -472,41 +472,65 @@ Because Windows privileged containers will work much differently than Linux priv
+- `$CONTAINER_SANDBOX_MOUNT_POINT` will not be set for non-`hostProcess` containers.
 - Volume mounts (including service account tokens) will be supported for privileged containers and will be mounted under the container volume. Programs running inside the container can either access volume mounts be using a relative path or by prefixing `$CONTAINER_SANDBOX_MOUNT_POINT` to their paths (example: use either `.\var\run\secrets\kubernetes.io\serviceaccount\` or `$CONTAINER_SANDBOX_MOUNT_POINT\var\run\secrets\kubernetes.io\serviceaccount\` to access service account tokens). These relative paths will be based on `Pod.containers.volumeMounts.mountPath`.
-- Client libraries such as https://pkg.go.dev/k8s.io/client-go/rest#InClusterConfig may be updated to prefix paths with `$CONTAINER_SANDBOX_MOUNT_POINT` if the environment variable is set for Windows so these libraries will work in `hostProcess`containers. This will be re-evaluated when transistioning from `alpha` to `beta` as we get more feedback.
-Note: it is not possible to feature-gate this behavior in client libraries and because of this the functionality should not be added to client libraries after privileged containers while this feature is in `alpha`.
-- Named Pipe mounts will **not** be supported. Instead named pipes should access via their path on the host (\\\\.\\pipe\\*). Unix domain sockets mounts **will** be supported.
-- All other volume types supported for normal containers on Windows will work with privileged containers.
+  - Note: We are prototyping a new approach to how the file system is created for `hostProcess` containers that would present the filesystem in a similar manner to non-hostProcess containers running on Windows (`c:\` (trailing \ included) would be the root instead of `c:\c\<container id>\`).
+  This would make it so files from volume mounts would be accessible via relative paths (ex: `/foo.exe` instead of needing to specify `$CONTAINER_SANDBOX_MOUNT_POINT/foo.exe`)
+  HostProcess containers would still have full access to the host file-system and `$CONTAINER_SANDBOX_MOUNT_POINT` would continue to be set so that workloads which already access files from inside volume months using this environment variable would continue to work without modification.
+  https://github.com/microsoft/hcsshim/pull/1107 is tracking this exploratory work.
+  This functionality will most-likely not be ready during Kubernetes v1.23 and any changes made to how volume mounts work would be done before this features becomes stable.
+
+- Client libraries such as https://pkg.go.dev/k8s.io/client-go/rest#InClusterConfig may be updated to prefix paths with `$CONTAINER_SANDBOX_MOUNT_POINT` if the environment variable is set for Windows so these libraries will work in `hostProcess` containers. This will be re-evaluated when transitioning from `alpha` to `beta` as we get more feedback.
```

Comment:
is this behavior expected to be added to the definition of how in-cluster credential/config loading should work, and be included in required client capabilities (https://github.com/kubernetes/community/blob/master/contributors/design-proposals/api-machinery/csi-new-client-library-procedure.md#client-capabilities) and propagated out to all clients in addition to client-go (https://github.com/orgs/kubernetes-client/repositories)?

I'm pretty squeamish about adding this to the client-libraries when we can't gate it and it's unclear if envvar-based mount dir approach is needed long-term

Reply:
I don't think we'll want / need any additional client libraries beyond the client-go library to be aware of this environment variable.

Reply:
Would you be OK with allowing only client-go library to prefix paths with this environment variable for 1.23. We did need that functionality in the client-go library to get to the point where we can deploy a CNI (calico) with pods on Windows.
We've proved out a number of our other top use cases for hostProcess containers and so far none of the other uses cases have needed this functionality. In a few weeks we will have a better understand of if the above-mentioned exploratory work will be pursued.

Reply:
discussed offline, deferred the decision whether or not to add this to client libraries until the investigation into preserving mount paths in hostProcess containers is complete, to avoid adding conditional code to client libraries that will not be needed/desired long-term